src/lib/components/MuiGridList/components/MuiGridListTileBar/MuiGridListTileBar.js

```diff
@@ -16,17 +19,33 @@ const MuiGridListTileBar = ({
   title,
 }) => {
   const styles = Storyblok.arrayToMuiStyles(rootClass);
+  let renderAction;
+  if (actionIcon.length !== 0) {
+    switch (actionIcon[0].component) {
+      case 'MuiIconButton':
+        renderAction = <MuiIconButton {...actionIcon[0]} />;
+        break;
+      case 'MuiIconButtonRedirect':
+        renderAction = <MuiIconButtonRedirect {...actionIcon[0]} />;
+        break;
+      case 'MuiIconButtonDialog':
+        renderAction = <MuiIconButtonDialog {...actionIcon[0]} />;
+        break;
+      case 'MuiIconButtonHref':
+        renderAction = <MuiIconButtonHref {...actionIcon[0]} />;
+        break;
+      default:
+        renderAction = null;
+        break;
+    }
+  } else { renderAction = null; }
 
   return (
     <>
       <GridListTileBar
         title={title}
         subtitle={<span>{subtitle}</span>}
-        actionIcon={
-          actionIcon.map(icon => (
-            <MuiIconButtonRedirect {...icon} />
-          ))
-        }
+        actionIcon={renderAction}
```

Comment:
actionIcon={actionIcon[0] && createElement(components[actionIcon[0].component, {...actionIcon[0]})}

maybe something like that?